test/pipelinefinally_test.go

```diff
@@ -84,6 +85,11 @@ func TestPipelineLevelFinally_OneDAGTaskFailed_InvalidTaskResult_Failure(t *test
 		t.Fatalf("Failed to create Task producing task results: %s", err)
 	}
 
+	taskConsumingResultInParam := getSuccessTaskConsumingResultsInParam(t, namespace)
+	if _, err := c.TaskClient.Create(ctx, taskConsumingResultInParam, metav1.CreateOptions{}); err != nil {
+		t.Fatalf("Failed to create Task consuming task results in param: %s", err)
+	}
+
 	taskConsumingResult := getSuccessTaskConsumingResults(t, namespace)
```

Comment:
im finding this a bit hard to follow, i think mostly b/c the details of the tasks are hidden in the functions (i.e. it was confusing before this change!), im wondering if some comments could help explain, e.g. what is taskConsumingResultInParam and how is it different from taskConsumingResult and why do we need both

Reply:
refactored to reuse the same function, which tasks the params list as an argument